### ISSUE TITLE
JAWs can no longer read plain Text Widget's content.

### ISSUE THREAD

**Issue by ipainchaud** [flutter/flutter#122607]
JAWs is the most widely used Screen Reader by the Blind and Low vision community. Although JAWs used to be pretty well supported, lately a pretty important issue arose: Plain Text Widgets are no longer read / discoverable by JAWs.

## Steps to Reproduce

1. Load the following [DartPad Gist](https://gist.github.com/ipainchaud/4b5ee58f9f78d1dbbb9eac49f8700fd8)
2. Ensure you are using the Stable Channel (flutter version 3.7.7)
3. Click on the "run" button.
4. Enable JAWs.
5. Click on the footer on the left section to set the focus at a reasonable place.
6. Press "Tab" until the focus reaches the "A button which does catch JAWs focus" (note that JAWs reads the button text, recognizing that it is a button)
7. Press the "up" arrow.

**Expected results:** We should have JAWs's focus lock on and read every line of text, one by one every time the "up" button is pressed. To test JAWs with the expected result, just change the channel to "old channel" (flutter v 3.3.10).

**Actual results:** JAWs skips every pieces of text and goes to the first "previous" button which is, in that case: the close button (with a "x").

Please refer to the [DartPad Gist](https://gist.github.com/ipainchaud/4b5ee58f9f78d1dbbb9eac49f8700fd8) for a minimal code sample. DartPad's linter didn't flag any warning at the time of the creation of this issue.

https://user-images.githubusercontent.com/5298744/225051325-ced6197a-2305-4c48-ada2-d0597be4a30c.mp4

**Comment by huycozy** [comment]
Thanks for filing the issue, @ipainchaud. This issue is reproducible on the latest `stable` and `master` channels with JAWs reader.

### Issue additional reports
- The issue doesn't happen on Flutter `3.3.10`
- It works as expected with NVDA reader on the latest channels

### Demo 

| master `3.9.0-1.0.pre.185` ❌ | `3.3.10` ✅ |
| --------------- | --------------- |
<video src="https://user-images.githubusercontent.com/104349824/225248985-19354cfe-04f2-4e6a-9ecf-5a6ee70514b7.mp4"/> | <video src="https://user-images.githubusercontent.com/104349824/225248970-e41d0304-aeb8-49b2-9d04-22ffe395cb97.mp4"/>

✅: No Issue ❌: Issue reproduced


<details>
<summary>flutter doctor -v (stable & master)</summary>

```bash
[√] Flutter (Channel stable, 3.7.7, on Microsoft Windows [Version 10.0.19045.2604], locale en-US)
    • Flutter version 3.7.7 on channel stable at C:\WIP\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2ad6cd72c0 (33 hours ago), 2023-03-08 09:41:59 -0800
    • Engine revision 1837b5be5f
    • Dart version 2.19.4
    • DevTools version 2.20.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\AndroidSDK
    • Platform android-33, build-tools 33.0.0
    • ANDROID_HOME = C:\AndroidSDK
    • Java binary at: C:\Program Files\Android\Android Studio Electric Eel 2022.1.1\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.4.4)
    • Visual Studio at D:\DOWNLOADWORK\VS2022
    • Visual Studio Community 2022 version 17.4.33213.308
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio Electric Eel 2022.1.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[√] VS Code (version 1.76.0)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.60.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19045.2604]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 110.0.5481.180
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 110.0.1587.41

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.9.0-1.0.pre.185, on Microsoft Windows [Version 10.0.19045.2604], locale en-US)
    • Flutter version 3.9.0-1.0.pre.185 on channel master at C:\WIP\flutter_master
    ! Warning: `flutter` on your path resolves to C:\WIP\flutter\bin\flutter, which is not inside your current
      Flutter SDK checkout at C:\WIP\flutter_master. Consider adding C:\WIP\flutter_master\bin to the front of your
      path.
    ! Warning: `dart` on your path resolves to C:\WIP\flutter\bin\dart, which is not inside your current Flutter SDK
      checkout at C:\WIP\flutter_master. Consider adding C:\WIP\flutter_master\bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d43d4823 (2 hours ago), 2023-03-14 03:34:17 -0400
    • Engine revision 6069c8f885
    • Dart version 3.0.0 (build 3.0.0-322.0.dev)
    • DevTools version 2.22.2
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git"
      directly to perform update checks and upgrades.

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\AndroidSDK
    • Platform android-33, build-tools 33.0.0
    • ANDROID_HOME = C:\AndroidSDK
    • Java binary at: C:\Program Files\Android\Android Studio Electric Eel 2022.1.1\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.4.4)
    • Visual Studio at D:\DOWNLOADWORK\VS2022
    • Visual Studio Community 2022 version 17.4.33213.308
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio Electric Eel 2022.1.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[√] VS Code (version 1.76.0)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.60.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19045.2604]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 110.0.5481.180
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 110.0.1587.41

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

Maybe related to: https://github.com/flutter/flutter/issues/75881

**Comment by ipainchaud** [comment]
I don't want to pretend I know what a "P4" label mean, but I can't keep myself from wondering if this could mean that this issue was classified as a low priority issue. If this is the case, I'd like to raise a case for the Blind and Low vision community. This bug affects the most widely used Screen Reader out there and would affect the vast majority of Blind or Low Vision users of Flutter apps out there. 

I can't keep myself from having a bleeding heart for all those who won't be able to use any Flutter WEB app because of that bug. So, if this "P4" label means that this is a "low priority" issue, I'd like to ask you guys to reconsider this prioritization.

I love what you guys do! Please keep it up!

**Comment by yjbanov** [comment]
P4 is the highest priority level for issues that are not being actively worked on. As soon as we're done with active accessibility issues, we'll bump this one to P3 and start working on it. It is not an indicator of less importance. It's just that we're busy with other issues (see [recently fixed accessibility issues](https://github.com/flutter/flutter/issues?q=is%3Aissue+label%3Aplatform-web+label%3A%22a%3A+accessibility%22+is%3Aclosed+sort%3Aupdated-desc)).

**Comment by yjbanov** [comment]
Update: JAWS ignores `aria-label` on this kind of DOM structure:

```
<div role="text" aria-label="My label"></div>
```

However, it is happy if I create a child and move the `aria-label` to the child:

```
<div role="text">
  <span aria-label="My label"></span>
</div>
```

I don't know why it does this. Probably a bug in JAWS, but we'll have to work around it. I'm going to try the latter approach, but I'll need to make sure all other screen readers are happy with it too.

Here's something similar discussed on SO: https://stackoverflow.com/questions/42681485/div-aria-label-not-being-read-by-jaws

**Comment by erhadesso** [comment]
Hi @yjbanov 
Can you estimate approximately when you will integrate the fix? A few months or longer?

Thanks for your awesome work!

**Comment by yjbanov** [comment]
Testing a potential fix right now, but also busy with some work stuff that (unfortunately) does not include coding. Maybe another week? (famous last words)

**Comment by erhadesso** [comment]
> Testing a potential fix right now, but also busy with some work stuff that (unfortunately) does not include coding. Maybe another week? (famous last words)

Hi, is there anything new yet?

**Comment by erhadesso** [comment]
@yjbanov Hi, sorry to bother you, any news yet?

**Comment by OmniRafa** [comment]
Hello @yjbanov and all,
we have developed a larger project in Flutter and are currently experiencing the same problem. Our app is explicitly developed for people with a visual impairment, and this bug is a very big obstacle for us. In addition, the support of a screen reader will be mandatory in our country. So this bug would be a huge priority for us. It depends on whether we can release or not. 
We are therefore very concerned about the status of the bug.
If this bug cannot be solved, it would be a huge problem for the customer and for us. We explicitly chose Flutter as the technology for the project because we believe in Flutter as a technology. However, if this bug cannot be fixed, it would destroy the customer's trust in Flutter.

**Comment by voipworld** [comment]
any news, still same issue with flutter 3.16.X

**Comment by yjbanov** [comment]
Update: here's a PR with a fix - https://github.com/flutter/engine/pull/50794

**Comment by yjbanov** [comment]
Update: I'm reworking this again. JAWs rejected the feature request (https://github.com/FreedomScientific/standards-support/issues/759), and Safari is slowly removing support for `role="text"` unfortunately. Even though it's not the most efficient option, we'll have to output labels as `Text` DOM nodes. On the bright side, it simplifies the code.
